konnectivity-client/pkg/client/client_test.go

```diff
@@ -44,6 +46,10 @@ func TestMain(m *testing.M) {
 func TestDial(t *testing.T) {
 	defer goleak.VerifyNone(t, goleak.IgnoreCurrent())
 
+	metrics.Metrics.RegisterMetrics(prometheus.DefaultRegisterer)
```

Comment:
All of these tests are now cannot be run in parallel. Not sure if there is a way to mark the tests as requiring they run in serial but at a minimum we should have a comment to that effect.